zeus/optimizer/batch_size/server/router.py

```diff
@@ -0,0 +1,127 @@
+"""Zeus batch size optimizer server FAST API router."""
+
+import logging
+from uuid import UUID
+
+from fastapi import Depends, FastAPI, Response, status
+from fastapi.responses import JSONResponse
+from sqlalchemy.ext.asyncio import AsyncSession
+from zeus.optimizer.batch_size.common import (
+    GET_NEXT_BATCH_SIZE_URL,
+    REGISTER_JOB_URL,
+    REPORT_RESULT_URL,
+    JobConfig,
+    ReportResponse,
+    TrainingResult,
+)
+from zeus.optimizer.batch_size.server.config import settings
+from zeus.optimizer.batch_size.server.database.db_connection import get_db_session
+from zeus.optimizer.batch_size.server.exceptions import ZeusBSOServerBaseError
+from zeus.optimizer.batch_size.server.optimizer import ZeusBatchSizeOptimizer
+from zeus.optimizer.batch_size.server.services.service import ZeusService
+from zeus.util.logging import get_logger
+
+app = FastAPI()
+logger = get_logger(__name__)
+logging.basicConfig(level=logging.getLevelName(settings.log_level))
+
+
+@app.on_event("startup")
+def startup_hook():
+    """Startup hook."""
+    pass
+
+
+@app.post(
+    REGISTER_JOB_URL,
+    responses={
+        200: {"description": "Job is already registered"},
+        201: {"description": "Job is successfully registered"},
+    },
+    response_model=JobConfig,
+)
+async def register_job(
+    job: JobConfig,
+    response: Response,
+    db_session: AsyncSession = Depends(get_db_session),
+) -> JobConfig:
+    """Endpoint for users to register a job or check if the job is registered and configuration is identical."""
+    optimizer = ZeusBatchSizeOptimizer(ZeusService(db_session))
+    try:
+        created = await optimizer.register_job(job)
+        await db_session.commit()
+        if created:
+            # new job is created
+            response.status_code = status.HTTP_201_CREATED
+        else:
+            # job already exists
+            response.status_code = status.HTTP_200_OK
+        return job
```

Comment:
Initially planned to let the user know what is the configuration that is registered in server, since `JobSpec` doesn't have other states. I think I can return `JobState` here. Wdyt?

Reply:
Actually returning JobState is nice (embedding the job ID). The `bso` can just hold on to it in `bso.job_state`, which reflects information about the current training run. But then, I wish the `JobState` model has a bit more information that the user can find useful, like:
- What is the current recurrence number?
- What is the current trial number in this recurrence?
- What is the state of the initial pruning round?

This is a small usability improvement and feel free to just not do it.

Reply:
But BTW, is it easy to calculate the current trail/recurrence number?

Reply:
Not really. We can take the maximum `round_number` from all explorations during the pruning stage. 
For the current recurrence number, we probably can do it by returning the number of `measurement` we have. 
Not sure what you meant by state of the `initial pruning round` though.

Reply:
A failed training trail does not count as a recurrence. So for example,

```
XXOXOOXO
X = convergence failure
O = convergence success
```

is just four recurrences. But I'm guessing the number of measurements is 8 in the DB?

Reply:
Ah true. We can filter by `converged = True` However, it might get slow if the number of measurements blows up, but probably doesn't have that many recurrence..? Not sure.

Reply:
Meh let's forget about this. Let's not return JobState. It's minor.